src/javascripts/views/homeView.js

```diff
@@ -1,7 +1,85 @@
+import Template from "../templates/template";
 class HomeView {
     constructor() {
-        this.transactionListEl = document.querySelector(".transaction-list")
+        this.categoryEl = document.querySelector(".category-list");
         this.addBtnEl = document.querySelector(".btn-add-transaction");
+        this.summaryEl = document.querySelector(".summary");
+        this.showTransactionsBtnEl = document.querySelector(".transaction-title-text");
+        this.showSummaryBtnEl = document.querySelector(".summary-text");
+    }
+
+     //----- RENDERING -----//
+     renderTransactionList(transactions) {
+        // Create an object to store transactions by category
+        const transactionByCategory = {};
+
+        // Browse through all transactions
+        for (const transaction of transactions) {
+        // Get the category of the current transaction
+         const category = transaction.category;
```

Comment:
@HoangLiarPhuongLe Please review the tab style used for indentation and ensure that it's uniform throughout the codebase.